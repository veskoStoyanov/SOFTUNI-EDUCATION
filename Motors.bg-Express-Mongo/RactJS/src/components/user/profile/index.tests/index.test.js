import React from 'react';
import { Provider } from 'react-redux'
import { shallow } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk';
import {Moto, Profile, UnbanBtn, UserBtn} from '../../..';

const mockStore = configureMockStore([thunk]);

describe('redux should render corecttly!', () => {
    it('Profile should work properly!', () => {
        const store = mockStore({
            startup: { complete: false }
        });
        const wrapper = shallow(
            <Provider store={store}>
                <Profile />
            </Provider>
        )
        expect(wrapper).toMatchSnapshot();
    });

    it('UnbanBtn should work properly!', () => {
        const store = mockStore({
            startup: { complete: false }
        });
        const wrapper = shallow(
            <Provider store={store}>
                <UnbanBtn />
            </Provider>
        )
        expect(wrapper).toMatchSnapshot();
    });

    it('UserBtn should work properly!', () => {
        const store = mockStore({
            startup: { complete: false }
        });
        const wrapper = shallow(
            <Provider store={store}>
                <UserBtn />
            </Provider>
        )
        expect(wrapper).toMatchSnapshot();
    });

    it("checks for html changes in Moto", () => {
        const wrapper = shallow(<Moto />);
        expect(wrapper).toMatchSnapshot();
    });
 
})

