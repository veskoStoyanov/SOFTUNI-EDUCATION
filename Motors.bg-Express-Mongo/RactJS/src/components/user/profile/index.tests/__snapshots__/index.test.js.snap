// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux should render corecttly! Profile should work properly! 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(Component) />
</ContextProvider>
`;

exports[`redux should render corecttly! UnbanBtn should work properly! 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(UnbanBtn) />
</ContextProvider>
`;

exports[`redux should render corecttly! UserBtn should work properly! 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(UserBtn) />
</ContextProvider>
`;

exports[`redux should render corecttly! checks for html changes in Moto 1`] = `
<li
  className="list-group-item"
>
  <h3
    className="mb-1"
  />
  <div
    className="col-md-8"
  >
    <img
      alt="profilePic"
      className="img-fluid imgg"
    />
  </div>
  <span
    className="float-right"
  >
    <Connect(UserBtn) />
  </span>
</li>
`;
